### ISSUE TITLE
"/_matrix/identity/v2/account/register" endpoint fails for web/desktop clients, but not for android mobile clients

### ISSUE THREAD

**Issue by vmario89** [matrix-org/synapse#15217]
### Description

Hi,
i have a strange issue with Matrix which i am not able to find it's root cause. We run a federating server with ~70 members. The server is up for 3 years now and works basically well. Some issue i registered today, as i saw that 3pids do not work correctly in element deskop or web client, but in element android. Registered mail addresses do not appear because API error is thrown (status 404 unrecognized). I do not get it. 

### Steps to reproduce

I checked 

- homeserver.yaml against the refence configuration documentation
- the nginx reverse proxy settings
- the log files

### Homeserver

matrix.fablabchemnitz.de

### Synapse Version

1.78.0

### Installation Method

regular apt repository

### Database

postgresql  12+214ubuntu0.1

### Workers

Single process

### Platform

Ubuntu 18 LTS on a Netcup virtual server

### Configuration

nginx:

```
server {
    listen 80;
    listen [::]:80;
    server_name matrix.fablabchemnitz.de;
    return 301 https://$server_name$request_uri;
}
server {
    listen 443 ssl http2;
    listen [::]:443 ssl http2;
    listen 8448 ssl http2 default_server;
    listen [::]:8448 ssl http2 default_server;
    include /etc/nginx/ssl-config.conf;
    server_name matrix.fablabchemnitz.de;
    access_log /var/log/nginx/matrix.fablabchemnitz.de.access.log;
    error_log /var/log/nginx/matrix.fablabchemnitz.de.error.log;
    client_max_body_size 100M;
     location / {
        default_type "application/json";
        allow all;
	root /var/www/vhosts/matrix.fablabchemnitz.de/;
    }
	location ~ ^(/_matrix|/_synapse/client) {
        proxy_pass http://localhost:8008;
        proxy_set_header X-Forwarded-For $remote_addr;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header Host $host;
        client_max_body_size 50M;
	proxy_http_version 1.1;
    }
    location ^~ /.well-known/matrix/server {
      return 200 '{"m.server": "matrix.fablabchemnitz.de"}';
      add_header Content-Type application/json;
    }
    location ^~ /.well-known/matrix/client {
      return 200 '{"m.homeserver": {"base_url": "https://matrix.fablabchemnitz.de"},"im.vector.riot.jitsi": {"preferredDomain": "https://jitsi.stadtfabrikanten.org"}}';
      add_header Content-Type application/json;
      add_header "Access-Control-Allow-Origin" *;
    }
}
```

### Relevant log output

var/log/matrix-synapse/homeserver.log says:
```
2023-03-06 20:30:31,096 - synapse.http.server - 107 - INFO - POST-697533 - <XForwardedForRequest at 0x7fc1581b84f0 method='POST' uri='/_matrix/identity/v2/account/register' clientproto='HTTP/1.1' site='8008'> SynapseError: 404 - Unrecognized request
2023-03-06 20:30:31,098 - synapse.access.http.8008 - 460 - INFO - POST-697533 - 80.134.84.181 - 8008 - {None} Processed request: 0.002sec/-0.000sec (0.000sec, 0.000sec) (0.000sec/0.000sec/0) 59B 404 "POST /_matrix/identity/v2/account/register HTTP/1.1" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Element/1.11.23 Chrome/108.0.5359.215 Electron/22.2.0 Safari/537.36" [0 dbevts]
```

### Anything else that would be useful to know?

Element Desktop on Fedora 37 as well as https://element.fablabchemnitz.de do not show up registered email addresses, but the mobile client  (v1.5.26) on android does as exspected.

Element Desktop or Web Client:
![grafik](https://user-images.githubusercontent.com/5631071/223216557-cb358e0a-f18f-4b38-8cd3-a0cdca7574ce.png)
![grafik](https://user-images.githubusercontent.com/5631071/223215632-ccb48636-b3dd-4d94-a83f-ac35ed522a2a.png)

Element Android Client:
![grafik](https://user-images.githubusercontent.com/5631071/223216666-cb3698b7-052c-4154-a1fb-5da99f307519.png)

**Comment by vmario89** [comment]
My homeserver.yaml file looks like this:

```
pid_file: "/var/run/matrix-synapse.pid"
public_baseurl: https://matrix.fablabchemnitz.de/
soft_file_limit: 0
use_presence: true
listeners:
  - port: 8008
    tls: false
    bind_addresses: ['::1', '127.0.0.1']
    type: http
    x_forwarded: true
    resources:
      - names: [client, media, federation, consent]
        compress: false
tls_certificate_path: "/var/lib/matrix-synapse/fullchain.pem"
tls_private_key_path: "/var/lib/matrix-synapse/privkey.pem"
federation_client_minimum_tls_version: 1.2
acme:
tls_fingerprints: [{"sha256":"<<<REDACTED>>>"}]
database:
    name: psycopg2
    args:
        user: <<<REDACTED>>>
        password: <<<REDACTED>>>
        database: <<<REDACTED>>>
        host: localhost
        cp_min: 5
        cp_max: 10
event_cache_size: "10K"
log_config: "/etc/matrix-synapse/log.yaml"
rc_messages_per_second: 0.2
rc_message_burst_count: 10.0
federation_rc_window_size: 1000
federation_rc_sleep_limit: 10
federation_rc_sleep_delay: 500
federation_rc_reject_limit: 50
federation_rc_concurrent: 3
media_store_path: "/var/lib/matrix-synapse/media"
uploads_path: "/var/lib/matrix-synapse/uploads"
max_upload_size: "50M"
max_image_pixels: "32M"
dynamic_thumbnails: true
thumbnail_sizes:
- width: 32
  height: 32
  method: crop
- width: 96
  height: 96
  method: crop
- width: 320
  height: 240
  method: scale
- width: 640
  height: 480
  method: scale
- width: 800
  height: 600
  method: scale
url_preview_enabled: True
url_preview_ip_range_blacklist:
- '127.0.0.0/8'
- '::1/128'
max_spider_size: "10M"
recaptcha_public_key: "<<<REDACTED>>>"
recaptcha_private_key: "<<<REDACTED>>>"
enable_registration_captcha: true
recaptcha_siteverify_api: "https://www.google.com/recaptcha/api/siteverify"
turn_uris: [ "<<<REDACTED>>>", "<<<REDACTED>>>" ]
turn_shared_secret: "<<<REDACTED>>>"
turn_user_lifetime: 86400000
turn_allow_guests: False
enable_registration: True
registration_shared_secret: <<<REDACTED>>>
bcrypt_rounds: 12
allow_guest_access: True
default_identity_server: https://matrix.fablabchemnitz.de
trusted_third_party_id_servers:
    - matrix.org
    - vector.im
suppress_key_server_warning: true
auto_join_rooms:
    - "#hallo:matrix.fablabchemnitz.de"
autocreate_auto_join_rooms: true
enable_metrics: True
room_prejoin_state:
app_service_config_files: []
track_appservice_user_ips: False
expire_access_token: False
signing_key_path: "/etc/matrix-synapse/homeserver.signing.key"
old_signing_keys: {}
key_refresh_interval: "1d"
trusted_key_servers:
  - server_name: "matrix.org"
password_config:
   enabled: true
email:
   enable_notifs: true
   smtp_host: "<<<REDACTED>>>"
   smtp_port: 587
   smtp_user: "<<<REDACTED>>>"
   smtp_pass: "<<<REDACTED>>>"
   require_transport_security: true
   notif_from: "Your Friendly %(app)s homeserver <matrix.noreply@fablabchemnitz.de>"
   app_name: Matrix
   notif_for_new_users: True
   client_base_url: "https://element.stadtfabrikanten.org"
enable_group_creation: true
user_consent:
   template_dir: privacy_policy_templates
   version: 1.0
   server_notice_content:
     msgtype: m.text
     body: >-
       Um mit der Nutzung dieses Homeservers fortzufahren müssen die Hinweise zur Datenschutzerklärung zur Kenntnis genommen werden. Siehe %(consent_uri)s
   send_server_notice_to_guests: False
   block_events_error: >-
       Um mit der Nutzung dieses Homeservers fortzufahren müssen die Hinweise zur Datenschutzerklärung zur Kenntnis genommen werden. Siehe %(consent_uri)s
   require_at_registration: True
   policy_name: Datenschutzerklärung
form_secret: <<<REDACTED>>>
server_notices:
  system_mxid_localpart: notices
  system_mxid_display_name: "Server-Hinweise"
  system_mxid_avatar_url: "<<<REDACTED>>>"
  room_name: "Server-Hinweise"
alias_creation_rules:
    - user_id: "*"
      alias: "*"
      action: allow
```

**Comment by DMRobertson** [comment]
> /_matrix/identity/v2/account/register

This is a request that is meant to be served by an "identity server", not a homeserver. ([See spec here](https://spec.matrix.org/v1.6/identity-service-api/#post_matrixidentityv2accountregister)).

You have configured:

> [default_identity_server](https://matrix-org.github.io/synapse/latest/usage/configuration/config_documentation.html#default_identity_server): https://matrix.fablabchemnitz.de

which is wrong if you are not hosting an identity server like [Sydent](https://github.com/matrix-org/sydent) or [ma1sd](https://github.com/ma1uta/ma1sd) at `matrix.fablabchemnitz.de`.

It is surprising that Android exhibits different behaviour. I note that the Synapse docs say of this setting:

> The identity server which we suggest that clients should use when users log in on this server.
> 
> (By default, no suggestion is made, so it is left up to the client. This setting is ignored unless public_baseurl is also explicitly set.)

So I would guess that element android uses `matrix.org` or `vector.im` as its default identity server.

**Comment by DMRobertson** [comment]
TL;DR I'd try either:
- removing the `default_identity_server` line, or
- host an identity server on https://matrix.fablabchemnitz.de/ (harder).

**Comment by vmario89** [comment]
Hi. thanks for looking into that. Yes thats why i was confused about. I modified homeserver.yaml and now the error is gone, after i changed the identity server for my account in the settings, too. Now another console errror is thrown in the desktop client:

`rageshake.ts:73 Failed to get URL preview: M_UNKNOWN: MatrixError: [502] Requested file's content type not allowed for this operation: application/pdf (https://matrix.fablabchemnitz.de/_matrix/media/r0/preview_url?url=https%3A%2F%2Fimpetus4cs.eu%2Fwp-content%2Fuploads%2F2023%2F01%2FGuide-for-applicants-2023.pdf&ts=1677832920000)`

maybe some CORS issue?

**Comment by DMRobertson** [comment]
That is a "normal" error (it should really be considered a warning in the client IMO)—Synapse refused to generate a URL preview for a PDF.

I'm going to close this assuming there are no other problems with 3pids on your deployment. If there are, feel free to reopen or open a new issue.
